update-css/sw.js

```diff
@@ -0,0 +1,108 @@
+
+  event.waitUntil(caches.open(cacheName).then(function(cache) {
+    return cache.addAll([
+      'style-1.css',
+      'style-2.css',
+    ]).then(getCurrentCSSFilename).then(function(filename) {
+      // Cache whichever CSS file we currently want to serve as
+      // 'style.css'
+      return cache.match(filename).then(function(response) {
+        if (response) {
+          cache.put('style.css', response);
+        }
+      });
+    });
+  }));
+});
+
+self.addEventListener('activate', function() {
+  if (self.clients && self.clients.claim) {
+    self.clients.claim();
+  }
+});
+
+self.addEventListener('fetch', function(event) {
+  // Only process fetch requests for 'style.css'. Let everything
+  // else be handled normally.
+  if (event.request.url.lastIndexOf('style.css') === -1) {
+    return;
+  }
+
+  // Respond with the cached version of 'style.css' after initiating
+  // a network fetch to update our cache if necessary.
+  event.respondWith(caches.open(cacheName).then(function(cache) {
+    // Perform a network fetch of 'style.css' to see if there's an updated
+    // version available. If there is, replace our cached version with the
+    // updated one and use `postMessage` to notify the page that there
+    // is updated CSS available.
+    simulateStyleCSSNetworkFetch().then(function(response) {
+      // In real production code, we would use smarter heuristics to
+      // determine if the CSS file that we requested has actually been
+      // updated.
+      if (response && response.text) {
+        var clone = response.clone();
+        clone.text().then(function(networkText) {
+          cache.match('style.css').then(function(cachedResponse) {
+            if (cachedResponse && cachedResponse.text) {
+              cachedResponse.text().then(function(cacheText) {
+                if (cacheText !== networkText) {
+                  cache.put('style.css', response);
+                  // Inform any open pages that a new CSS file is available on
+                  // the server
+                  notifyPageOfCSSUpdate(true);
+                } else {
+                  notifyPageOfCSSUpdate(false);
+                }
+              });
+            } else {
+              notifyPageOfCSSUpdate(false);
+            }
+          });
+        });
+      } else {
+        notifyPageOfCSSUpdate(false);
+      }
+    });
+
+    return cache.match('style.css');
+  }));
+});
+
+function notifyPageOfCSSUpdate(isUpdated) {
+  self.clients.matchAll().then(function(clientList) {
+    clientList.forEach(function(client) {
+      client.postMessage({ msg: 'cssUpdated', val: isUpdated });
+    });
+  });
+}
+
+function simulateStyleCSSNetworkFetch() {
+  return getCurrentCSSFilename().then(function(filename) {
+    return caches.open(cacheName).then(function(cache) {
+      // return the specified CSS file from the cache
+      return cache.match(filename);
+    });
+  });
+}
+
+function getCurrentCSSFilename() {
```

Comment:
If the _cache-as-db_ trick is an implementation detail I would hide that implementation detail. When solving #14, in #39, we restricted docco to the js of the first level, so you could use `importScripts(...)` and create something like `get().then()` and `put().then()` to actually hide these details (the we can open a followup to implement this with IndexedDB).